### PR TITLE
enabling reseting autodiff_Values instead of recreating to reduce overhead

### PR DESCRIPTION
**Pull request by lquante**

### CODE REVIEW

File: autodiff.h [hunk 1]
```diff
@@ -82,6 +82,12 @@ class Value final {
     const Vector& derivative() const { return dev; }
     explicit operator T() const { return val; }
 
+    Value reset() {
```
Comment (swillner) [inline autodiff.h:85]: would just keep it a `void` method

File: autodiff.h [hunk 2]
```diff
@@ -82,6 +82,12 @@ class Value final {
...
+        val = val * 0;  // most general, since scalar values are also possible
```
Comment (swillner) [inline autodiff.h:86]: why not `val = 0`?

**Review by swillner — CHANGES_REQUESTED** [review]

File: autodiff.h (hunk 2 above)
Comment (lquante) [reply autodiff.h:86]: ok

File: autodiff.h (hunk 1 above)
Comment (lquante) [reply autodiff.h:85]: sure

**Review by swillner — APPROVED** [review]

**Pull request merged**